lib/OpenQA/CacheService.pm

```diff
@@ -130,9 +130,16 @@ sub run {
     $ENV{MOJO_INACTIVITY_TIMEOUT} //= 300;
     $app->log->debug("Starting cache service: $0 @args");
     $app->defaults->{service_pid} = $$;
-
-    my $cmd_return_code = $app->start(@args);
-    return $app->exit_code // $cmd_return_code // 0;
+    my $e;
+    my $cmd_return_code;
+    do {
+        eval { $cmd_return_code = $app->start(@args) };
+        chomp($e = $@);
+        return $app->exit_code // $cmd_return_code // 0 unless $e;
+        my $retry_interval = $ENV{OPENQA_CACHE_SERVICE_RETRY_INTERVAL} // 10;
+        $app->log->info("cache service failed with '$e', retrying after $retry_interval seconds");
```

Comment:
It'll actually only retry when `$e =~ /Cannot assign requested address/` so logging the message always is a bit confusing.

Reply:
If the error message is *not* matching `Cannot assign requested address` it should likely be logged as an error message.

Reply:
Done both I think by calling die in the line above unless we have the specific message